### ISSUE TITLE
[LOVE] Add a cpu count fact

### ISSUE THREAD

**Issue by purpleidea** [purpleidea/mgmt#398]
#### Welcome:

This is an intermediate to advanced level patch for someone who is at least an intermediate golang user, but being new to mgmt is okay. It is meant to encourage new programmers to get involved with mgmt. It will probably take you at least a day to complete.

#### Intro:

Mgmt has a number of built-in variables ("facts") in the language. They are here:
https://github.com/purpleidea/mgmt/tree/master/lang/funcs/facts/core
We'd like to add a new one for CPU count. It should change when the cpu count changes (hotplug/hotunplug).

#### Design:

It seems it's possible to get udev events from the cpu count changes. It's easiest to test this inside of a libvirt vm, where hotplug/hotunplug works reliably. (Even via the mgmt `virt` resource.)
Some helpful links:
https://people.freedesktop.org/~narmstrong/meson_drm_doc/core-api/cpu_hotplug.html#user-space-notification
We should consider using udev via some udev golang library. You'll need to decide which one to use (if it is required) and as well find a method to reliably get the cpu count too. (You still need to get an initial value when there are no events.)
A value of zero could represent an error in reading the cpu count value.
Once you've decided on a design, present it here for discussion before you implement things.

Read the contributing tips:
https://github.com/purpleidea/mgmt/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Happy hacking, and ping me for questions!

#### Bonus/Future work:

* Add some tests!
* Add an example to the github repo in examples/lang/.
* Discuss whether it makes sense to expose other properties of the cpu's in a fact. Perhaps a struct of values instead of just an int?

**Comment by purpleidea** [comment]
PS: I found this quick snippet: It would be an easy test to see if it works without needing lots of deps: https://gist.github.com/rmohr/23b7ae12ac0b8d0201a5

**Comment by keur** [comment]
I'll take this. Will focus on it  after #430 is figured out

**Comment by purpleidea** [comment]
Here is a brief analysis of this one lib: https://github.com/jochenvg/go-udev/issues/7

**Comment by purpleidea** [comment]
This lovely feature has been added thanks to @keur, so closing!

In the future we could always consider returning a `struct` with a bunch of different info about the cpus, but I don't think it's necessary at the moment. Should be fun to build some stuff with this, and we learned a bunch about sockets, and even fixed a race in `net`!
